src/python_testing/TC_CADMIN_1_11.py

```diff
@@ -0,0 +1,197 @@
+import random
+from time import sleep
+
+import chip.clusters as Clusters
+from chip import ChipDeviceCtrl
+from chip.ChipDeviceCtrl import CommissioningParameters
+from chip.exceptions import ChipStackError
+from matter_testing_support import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+from mobly import asserts
+
+
+class TC_CADMIN_1_11(MatterBaseTest):
+    async def OpenCommissioningWindow(self, th, expectedErrCode) -> CommissioningParameters:
+        if expectedErrCode == 0x00:
+            params = await th.OpenCommissioningWindow(
+                nodeid=self.dut_node_id, timeout=self.timeout, iteration=10000, discriminator=self.discriminator, option=1)
+            return params
+
+        elif expectedErrCode == 0x02:
+            ctx = asserts.assert_raises(ChipStackError)
+            with ctx:
+                await th.OpenCommissioningWindow(
+                    nodeid=self.dut_node_id, timeout=self.timeout, iteration=10000, discriminator=self.discriminator, option=1)
+            errcode = ctx.exception.chip_error
+            logging.info('Commissioning complete done. Successful? {}, errorcode = {}'.format(errcode.is_success, errcode))
+            asserts.assert_false(errcode.is_success, 'Commissioning complete did not error as expected')
+            asserts.assert_true(errcode.sdk_code == expectedErrCode,
+                                'Unexpected error code returned from CommissioningComplete')
+
+    async def OpenBasicCommissioningWindow(self, th, expectedErrCode) -> CommissioningParameters:
+        if expectedErrCode == 0x00:
+            params = await th.OpenBasicCommissioningWindow(
+                nodeid=self.dut_node_id, timeout=self.timeout)
+            return params
+
+        elif expectedErrCode == 0x02:
+            ctx = asserts.assert_raises(ChipStackError)
+            with ctx:
+                await th.OpenBasicCommissioningWindow(
+                    nodeid=self.dut_node_id, timeout=self.timeout)
+            errcode = ctx.exception.chip_error
+            logging.info('Commissioning complete done. Successful? {}, errorcode = {}'.format(errcode.is_success, errcode))
+            asserts.assert_false(errcode.is_success, 'Commissioning complete did not error as expected')
+            asserts.assert_true(errcode.sdk_code == expectedErrCode,
+                                'Unexpected error code returned from CommissioningComplete')
+
+    async def read_currentfabricindex(self, th: ChipDeviceCtrl) -> int:
+        cluster = Clusters.Objects.OperationalCredentials
+        attribute = Clusters.OperationalCredentials.Attributes.CurrentFabricIndex
+        current_fabric_index = await self.read_single_attribute_check_success(dev_ctrl=th, endpoint=0, cluster=cluster, attribute=attribute)
+        return current_fabric_index
+
+    def steps_TC_CADMIN_1_11(self) -> list[TestStep]:
+        return [
+            TestStep(1, "Commissioning, already done", is_commissioning=True),
```

Comment:
could you please add the expectations in here as well?

Reply:
Hi Cecille,
I have added the expectations to the Test steps, hopefully the ones added are acceptable.